pkg/kubelet/kubelet.go

```diff
@@ -157,7 +157,8 @@ type Kubelet struct {
 	podInfraContainerImage string
 	podWorkers             *podWorkers
 	resyncInterval         time.Duration
-	pods                   []api.BoundPod
+	pods                   []api.BoundPod // guarded by podsMutex
+	podsMutex              sync.RWMutex
```

Comment:
I noticed that I missed that one of the function is returning pointer. I'll need to handle that as well.